src/envelope.rs

```diff
@@ -10,30 +10,32 @@ pub(crate) struct Envelope {
     pub(crate) message: Protocol,
 }
 
+/// Supported network protocols.
 #[derive(Debug)]
-pub(crate) enum Protocol {
+pub enum Protocol {
     Tcp(Segment),
     Udp(Datagram),
 }
 
+/// UDP datagram.
 #[derive(Debug)]
-pub(crate) struct Datagram(pub Bytes);
+pub struct Datagram(pub Bytes);
 
 /// This is a simplification of real TCP.
 ///
 /// We implement just enough to ensure fidelity and provide knobs for real world
 /// scenarios, but we skip a ton of complexity (e.g. checksums, flow control,
 /// etc) because said complexity isn't useful in tests.
 #[derive(Debug)]
-pub(crate) enum Segment {
+pub enum Segment {
     Syn(Syn),
     Data(u64, Bytes),
     Fin(u64),
     Rst,
 }
 
 #[derive(Debug)]
-pub(crate) struct Syn {
+pub struct Syn {
```

Comment:
Is there any reason you prefer `pub(crate)` to `pub` with visibility limited to crate at the module declaration?

Reply:
No reason, just started out this way. We could go back and and fix this at some point for the internals.